cdap-kubernetes/src/main/java/io/cdap/cdap/master/environment/k8s/KubeMasterEnvironment.java

```diff
@@ -389,46 +409,67 @@ private V1PodSpec getSparkPodSpec(PodInfo podInfo) {
+    try {
+      List<V1Volume> volumes = new ArrayList<>();
+      V1DownwardAPIVolumeFile labelsFile = new V1DownwardAPIVolumeFile()
+        .fieldRef(new V1ObjectFieldSelector().fieldPath("metadata.labels"))
+        .path(podLabelsFile.getName());
+      V1DownwardAPIVolumeFile nameFile = new V1DownwardAPIVolumeFile()
+        .fieldRef(new V1ObjectFieldSelector().fieldPath("metadata.name"))
+        .path(podNameFile.getName());
+      V1DownwardAPIVolumeFile uidFile = new V1DownwardAPIVolumeFile()
+        .fieldRef(new V1ObjectFieldSelector().fieldPath("metadata.uid"))
+        .path(podUidFile.getName());
+      V1DownwardAPIVolumeSource podinfoVolume = new V1DownwardAPIVolumeSource()
+        .defaultMode(420)
+        .items(Arrays.asList(labelsFile, nameFile, uidFile));
+      String podInfoVolumeName = "podinfo";
+      volumes.add(new V1Volume().downwardAPI(podinfoVolume).name(podInfoVolumeName));
+
+      List<V1VolumeMount> volumeMounts = new ArrayList<>();
+      volumeMounts.add(new V1VolumeMount().name(podInfoVolumeName).mountPath(podInfoDir.getAbsolutePath()));
+
+      // mount spark localize resources
+      V1ConfigMapBuilder configMapBuilder = new V1ConfigMapBuilder();
+      V1ConfigMapVolumeSourceBuilder configVolumeBuilder = new V1ConfigMapVolumeSourceBuilder();
+
+      for (Map.Entry<String, SparkLocalizeResource> resource : localizeResources.entrySet()) {
+        // TODO: CDAP-18504 Make this binary data when changing to compressed files
+        configMapBuilder.addToData(resource.getKey(), readFileContents(resource.getValue().getURI()));
+        configVolumeBuilder.addToItems(new V1KeyToPath().key(resource.getKey()).path(resource.getKey()));
       }
+
+      // Create config map
+      String configMapName = CDAP_CONFIG_MAP_PREFIX + UUID.randomUUID();
+      coreV1Api.createNamespacedConfigMap(podInfo.getNamespace(),
+                                          configMapBuilder.withMetadata(
+                                            new V1ObjectMetaBuilder()
+                                              .withName(configMapName).build()).build(),
+                                          null, null, null);
+      this.configMapName = configMapName;
+
+      // Create configmap Volume and Volume mount to be added to the pod template
+      volumes.add(new V1Volume().name(configMapName).configMap(configVolumeBuilder.withName(configMapName).build()));
+      volumeMounts.add(new V1VolumeMount().name(configMapName).mountPath(CDAP_LOCALIZE_FILES_PATH).readOnly(true));
+
+      v1PodSpecBuilder
+        .withVolumes(volumes)
+        .withContainers(new V1ContainerBuilder()
+                          .withVolumeMounts(volumeMounts)
+                          .build());
+
+    } catch (ApiException e) {
+      throw new RuntimeException("Error occurred while creating pod spec. Error code="
+                                   + e.getCode() + ", Body=" + e.getResponseBody(), e);
+    } catch (Exception e) {
```

Comment:
Are there other types of checked exceptions that can be thrown? In general we should avoid wrapping checked exceptions in runtime exceptions.

Reply:
Hmm. Thats a good point. Since this is a get method, I didnt add an exception to the method signature. But now the get method does some mutations on master environment (create configmap).
Let me rename the method and add `Exception` to method signature for the caller.